src/states/elder/mod.rs

```diff
@@ -2087,6 +2059,12 @@ impl Approved for Elder {
         Ok(())
     }
 
+    fn handle_ack_message_event(&mut self, sec_info: SectionInfo) -> Result<(), RoutingError> {
+        self.chain
+            .update_their_knowledge(*sec_info.prefix(), *sec_info.version());
```

Comment:
Is `sec_info.version()` correct here? We want the `AckMessage` to point not to a SectionInfo, but to a SectionProofBlock within `our_history`. The index in `our_history` and the version in `SectionInfo` might differ.

What we should be doing, in my opinion, would be to put a `BlsPublicKey` in the `AckMessage` (which would ideally be taken from `security_metadata` of the original message, but that's what I'm working on at the moment), then on receiving `AckMessage` find the block that contains this key and update `their_knowledge` to the index of this block.

Since `security_metadata` is not yet done, we could temporarily do this thing (and maybe add a TODO), or alternatively we'd have to postpone this until we have `security_metadata` and a proper source of the public key to vote on.

Reply:
> The index in our_history and the version in SectionInfo might differ.

True, but that does not seem to be the their_knowledge job to make that adjustment, otherwise we need to update that map every time we prune our_history.

> What we should be doing, in my opinion...
Seem to be functionaly equivalent until the part were we use the block index which may change, and that does not seem nice.

Reply:
Issue raised: https://github.com/maidsafe/routing/issues/1722